tests/unit/python/foglamp/plugins/south/http_south/test_http_south.py

```diff
@@ -0,0 +1,294 @@
+from foglamp.plugins.south.http_south.http_south import HttpSouthIngest
+from foglamp.plugins.south.coap_listen.coap_listen import Ingest
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_CONFIG_CATEGORY_NAME = 'HTTP_SOUTH'
+_CONFIG_CATEGORY_DESCRIPTION = 'South Plugin HTTP Listener'
+
+
+def _DEFAULT_CONFIG(unused_port):
+    return {
+        'plugin': {
+            'description': 'South Plugin HTTP Listener',
+            'type': 'string',
+            'default': 'http_south'
+        },
+        'port': {
+            'description': 'Port to listen on',
+            'type': 'integer',
+            'default': str(unused_port),
+        },
+        'host': {
+            'description': 'Address to accept data on',
+            'type': 'string',
+            'default': '0.0.0.0',
+        },
+        'uri': {
+            'description': 'URI to accept data on',
+            'type': 'string',
+            'default': 'sensor-reading',
+        }
+    }
+
+
+_NEW_CONFIG = {
+    'plugin': {
+        'description': 'South Plugin HTTP Listener',
+        'type': 'string',
+        'default': 'http_south'
+    },
+    'port': {
+        'description': 'Port to listen on',
+        'type': 'integer',
+        'default': '1234',
+    },
+    'host': {
+        'description': 'Address to accept data on',
+        'type': 'string',
+        'default': 'localhost',
+    },
+    'uri': {
+        'description': 'URI to accept data on',
+        'type': 'string',
+        'default': 'sensor-reading',
+    }
+}
+
+
+def mock_request(data, loop):
+    payload = StreamReader(loop=loop)
+    payload.feed_data(data.encode())
+    payload.feed_eof()
+
+    protocol = mock.Mock()
+    app = mock.Mock()
+    headers = CIMultiDict([('CONTENT-TYPE', 'application/json')])
+    req = make_mocked_request('POST', '/sensor-reading', headers=headers,
+                              protocol=protocol, payload=payload, app=app)
+    return req
+
+
+def test_plugin_info():
+    assert http_south.plugin_info() == {
+        'name': 'http_south',
+        'version': '1.0',
+        'mode': 'async',
+        'type': 'south',
+        'interface': '1.0',
+        'config': _DEFAULT_CONFIG(6683)
+    }
+
+
+def test_plugin_init(unused_port):
+    config = _DEFAULT_CONFIG(unused_port())
+    assert http_south.plugin_init(config) == config
+
+
+def test_plugin_start(unused_port):
+    # GIVEN
+    config = copy.deepcopy(_DEFAULT_CONFIG(unused_port()))
+    config['port']['value'] = config['port']['default']
+    config['host']['value'] = config['host']['default']
+    config['uri']['value'] = config['uri']['default']
+
+    # WHEN
+    loop = asyncio.get_event_loop()
+    loop.call_soon(http_south.plugin_start(config))
+
+    # THEN
+    assert config['app'] is not None
+    assert config['handler'] is not None
+    assert config['server'] is not None
```

Comment:
assert is not None is not a good assertion, suggest to have definitive assertions like:
```
assert isinstance(config['app'], aiohttp.web.Application)
assert isinstance(config['handler'], aiohttp.web_server.Server)
assert isinstance(config['server'], asyncio.base_events.Server)
```